### PR TITLE
kernel: Add TPM Modules support

### PR DESCRIPTION
**Pull request by riptidewave93**
The following will enable the TPM kernel module, as well as support for the atmel i2c TPM driver. Tested and confirmed working on an Aerohive AP-121

Signed-off-by: Chris Blake <chrisrblake93@gmail.com>

### CODE REVIEW

File: package/kernel/linux/modules/other.mk [hunk 1]
```diff
@@ -954,3 +954,56 @@ define KernelPackage/bmp085-spi/description
...
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=TPM Hardware Support
+  KCONFIG:= CONFIG_TCG_TPM \
+	CONFIG_TCG_TIS=n \
+	CONFIG_TCG_TIS_I2C_ATMEL=n \
+	CONFIG_TCG_TIS_I2C_INFINEON=n \
+	CONFIG_TCG_TIS_I2C_NUVOTON=n \
+	CONFIG_TCG_ATMEL=n \
+	CONFIG_TCG_TIS_ST33ZP24=n
+  FILES:= $(LINUX_DIR)/drivers/char/tpm/tpm.ko
```
Comment (blogic) [inline package/kernel/linux/modules/other.mk]: can you please add the =n lines into the generic config files please and drop them from this file.

**Comment by riptidewave93** [comment]
Requested changes have been applied.

**Comment by blogic** [comment]
merged, Thanks for the quick response !

**Comment by nbd168** [comment]
Next time you add a kernel module package, please test the target/compile step on every major kernel version with CONFIG_ALL_KMODS=y

**Comment by riptidewave93** [comment]
@nbd168 Sorry about that, will be sure to do that for any future kernel modules/changes. Thanks for the feedback.

**Pull request closed without merging**